tests/cnf/ran/ztp/tests/ztp-spoke-checker.go

```diff
@@ -6,40 +6,66 @@ import (
 	"github.com/golang/glog"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
 	"github.com/openshift-kni/eco-goinfra/pkg/pod"
 	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
 	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/cluster"
 	. "github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
 	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/ztp/internal/tsparams"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
 )
 
 var _ = Describe("ZTP Spoke Checker Tests", Label(tsparams.LabelSpokeCheckerTests), func() {
+	var nodeSelector metav1.ListOptions
+
+	BeforeEach(func() {
+		nodeList, err := nodes.List(Spoke1APIClient)
+		Expect(err).ToNot(HaveOccurred(), "Failed to list nodes for spoke 1")
+		Expect(nodeList).ToNot(BeEmpty(), "Failed to find any node for spoke 1")
+
+		nodeSelector = metav1.ListOptions{
+			FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": nodeList[0].Definition.Name}).String(),
```

Comment:
Perhaps we should check all nodes instead of the first node. As in the case of sno+worker, the two nodes are not equal, they got installed differently, so it would be nice to check both with a loop inside the test function?

Reply:
@yliu127 I originally changed it to use all nodes but it seemed to fail running the command on the worker node and it looks like the worker node does not have the services disabled. The cnf-gotests version just checks the first node which seems to always be the control-plane